core/src/test/resources/org/apache/calcite/test/SqlToRelConverterTest.xml

```diff
@@ -323,7 +323,121 @@ LogicalProject(DEPTNO=[$0], NAME=[$1], I=[$2])
 ]]>
         </Resource>
     </TestCase>
-    <TestCase name="testSnapshotOnTemporalTable1">
+    <TestCase name="testTableFunctionInSelect">
+        <Resource name="sql">
+            <![CDATA[select * from (select table_func(dept.deptno) as (f0, f1) from dept)]]>
+        </Resource>
+        <Resource name="plan">
+            <![CDATA[
+LogicalProject(F0=[$2], F1=[$3])
+  LogicalCorrelate(correlation=[$cor0], joinType=[inner], requiredColumns=[{0}])
+    LogicalTableScan(table=[[CATALOG, SALES, DEPT]])
+    LogicalTableFunctionScan(invocation=[TABLE_FUNC($cor0.DEPTNO)], rowType=[RecordType(BIGINT f0, VARCHAR f1)], elementType=[class [Ljava.lang.Object;])
```

Comment:
The plan is not correct, there is no need to translate as a `LogicalCorrelate`, the `LogicalTableFunctionScan` inputs need to be set correctly.

Reply:
I am not sure whether it should use `LogicalCorrelate ` or not. Some existed examples use `LogicalCorrelate `[1]. Any suggestion, @julianhyde ?

[1] https://github.com/apache/calcite/blob/master/core/src/test/resources/org/apache/calcite/test/SqlToRelConverterTest.xml#L320

Reply:
I think `LoggicalCorrelate` is the right thing. This is a function that RETURNS a table on each invocation but its arguments are scalars, and the scalars are correlating variables.

Reply:
If we just select from a function using `select * from func`, i think there is no need to translate into a `LoggicalCorrelate `, because usually a `LoggicalCorrelate` would be decorrelated into a `Join`, can we elaborate the `LogicalTableFunctionScan` inputs to specify the input table and function's argument as an `input ref` ?

Reply:
When would we have `select * from func`? We only have `select func(a, b) as (a. b) from t`. If I am worong, correct me please.

Reply:
For query `select func(a, b) as (a. b) from t`, we can generate a `LogicalTableFunctionScan` directly instead of a `Correlate` plus a right project.

Reply:
OK. I would like to see if we can make it.